mitodl/bootcamp-ecommerce
# Refund enrollments command

#### What are the relevant tickets?
Partially addresses #592 

#### What's this PR do?
Creates a management command to issue a refund for an enrollment.

#### How should this be manually tested?
- Apply for a bootcamp and make a full payment 
- In django admin, run `ecommerce.api.complete_successful_order(order)` on the order you made.  There should now be a `BootcampEnrollment` object afterward.
- Run this command and enter an amount greater than what you paid, you should get an error indicating that the refund amount exceeds the amount paid.:
```python manage.py refund_enrollment --user <username> --run <bootcamp_run.id> --amount <too_big_amount>```
- Run this command again but enter an amount that is half of what you paid, the command should complete successfully and there should be a new fulfilled `Order` with a payment amount equal to the negative of what you entered.
- Run it again for the same amount, it should succeed again and there should be an additional order.
- Run it one more time, it should fail with an error indicating that the refund amount exceeds the amount paid.

#### Any background context you want to provide?
- Entering an amount is necessary because some bootcamps will have non-refundable deposits that are not stored in the database.
- The code needs to handle refunds for enrollments that may not have applications (which may be possible in the future via bulk purchases).
